gnark-prover/prover/poseidon/constants.go

```diff
@@ -497,4 +498,417 @@ var CONSTANTS_3 = [][]frontend.Variable{
 		hex("0x216f6717bbc7dedb08536a2220843f4e2da5f1daa9ebdefde8a5ea7344798d22"),
 		hex("0x1da55cc900f0d21f4a3e694391918a1b3c23b2ac773c6b3ef88e2e4228325161"),
 	},
-}
\ No newline at end of file
+}
+
+var MDS_4 = [][]frontend.Variable{
+	{
+		hex("0x236d13393ef85cc48a351dd786dd7a1de5e39942296127fd87947223ae5108ad"),
+		hex("0x277686494f7644bbc4a9b194e10724eb967f1dc58718e59e3cedc821b2a7ae19"),
+		hex("0x023db68784e3f0cc0b85618826a9b3505129c16479973b0a84a4529e66b09c62"),
+		hex("0x1d359d245f286c12d50d663bae733f978af08cdbd63017c57b3a75646ff382c1"),
+	},
+	{
+		hex("0x2a75a171563b807db525be259699ab28fe9bc7fb1f70943ff049bc970e841a0c"),
+		hex("0x083abff5e10051f078e2827d092e1ae808b4dd3e15ccc3706f38ce4157b6770e"),
+		hex("0x1a5ad71bbbecd8a97dc49cfdbae303ad24d5c4741eab8b7568a9ff8253a1eb6f"),
+		hex("0x0d745fd00dd167fb86772133640f02ce945004a7bc2c59e8790f725c5d84f0f"),
+	},
+	{
+		hex("0x2070679e798782ef592a52ca9cef820d497ad2eecbaa7e42f366b3e521c4ed42"),
+		hex("0x2e18c8570d20bf5df800739a53da75d906ece318cd224ab6b3a2be979e2d7eab"),
+		hex("0x0fa86f0f27e4d3dd7f3367ce86f684f1f2e4386d3e5b9f38fa283c6aa723b608"),
+		hex("0x03f3e6fab791f16628168e4b14dbaeb657035ee3da6b2ca83f0c2491e0b403eb"),
+	},
+	{
+		hex("0x2f545e578202c9732488540e41f783b68ff0613fd79375f8ba8b3d30958e7677"),
+		hex("0x23810bf82877fc19bff7eefeae3faf4bb8104c32ba4cd701596a15623d01476e"),
+		hex("0x014fcd5eb0be6d5beeafc4944034cf321c068ef930f10be2207ed58d2a34cdd6"),
+		hex("0x00c15fc3a1d5733dd835eae0823e377f8ba4a8b627627cc2bb661c25d20fb52a"),
+	},
+}

```

Comment:
mds matrix parameters seem to be in a different order than circom lib and there are some inconsistencies as well.
see https://github.com/iden3/circomlib/blob/cff5ab6288b55ef23602221694a6a38a0239dcc0/circuits/poseidon_constants.circom#L2334-L2345